asserts/repair.go

```diff
@@ -0,0 +1,128 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+	"time"
+)
+
+// Repair holds an repair assertion which allows running repair
+// code to fixup broken systems. It can be limited by series and models.
+type Repair struct {
+	assertionBase
+
+	series []string
+	models []string
+	since  time.Time
+	until  time.Time
+}
+
+// RepairID returns the "id" of the repair. It should be a short string
+// that follows a convention like "REPAIR-123". Similar to a CVE there
+// should be a public place to look up details about the repair-id
+// (e.g. the snapcraft forum).
+func (em *Repair) RepairID() string {
+	return em.HeaderString("repair-id")
+}
+
+// Series returns the series that this assertion is valid for.
+func (em *Repair) Series() []string {
+	return em.series
+}
+
+// Models returns the models that this assertion is valid for.
+func (em *Repair) Models() []string {
+	return em.models
```

Comment:
If you go ahead with a wildcard support to models we should have a function that checks if a model needs a given repair.